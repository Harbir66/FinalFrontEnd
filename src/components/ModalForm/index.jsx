import React from 'react';
import './ModalForm.css';
import PropTypes from 'prop-types';

function ModalForm({ isOpen, onClose, children }) {
  if (!isOpen) return null;

  return (
    <div className="modal">
      <div className="modal-overlay" onClick={onClose} />
      <div className="modal-content">{children}</div>
    </div>
  );
}

export default ModalForm;

ModalForm.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  onClose: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
};
