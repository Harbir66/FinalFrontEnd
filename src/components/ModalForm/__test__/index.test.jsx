import { render } from '@testing-library/react';
import React from 'react';
import ModalForm from '..';

describe('ModalForm', () => {
  it('should render when not open', () => {
    const { asFragment } = render(<ModalForm />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should render when open', () => {
    const { asFragment } = render(<ModalForm isOpen onClose={jest.fn()} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
