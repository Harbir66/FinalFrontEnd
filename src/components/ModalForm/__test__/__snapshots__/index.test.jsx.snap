// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalForm should render when not open 1`] = `<DocumentFragment />`;

exports[`ModalForm should render when open 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <div
      class="modal-overlay"
    />
    <div
      class="modal-content"
    />
  </div>
</DocumentFragment>
`;
